src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/incentives/IncentivesReferenceService.kt

```diff
@@ -0,0 +1,45 @@
+package uk.gov.justice.digital.hmpps.prisonertonomisupdate.incentives
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.microsoft.applicationinsights.TelemetryClient
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.prisonertonomisupdate.services.NomisApiService
+import uk.gov.justice.digital.hmpps.prisonertonomisupdate.services.ReferenceCode
+
+@Service
+class IncentivesReferenceService(
+  private val incentivesApiService: IncentivesApiService,
+  private val nomisApiService: NomisApiService,
+  private val incentivesUpdateQueueService: IncentivesUpdateQueueService,
+  private val telemetryClient: TelemetryClient,
+  private val objectMapper: ObjectMapper,
+) {
+
+  private fun IncentiveLevel.toNomisIncentiveLevel(): ReferenceCode = ReferenceCode(
+    code = code,
+    domain = "IEP_LEVEL",
+    description = description,
+    active = active,
+  )
+
+  suspend fun globalIncentiveLevelChange(event: GlobalIncentiveChangedEvent) {
+    val incentiveLevelCode = event.additionalInformation.incentiveLevel
+    nomisApiService.getGlobalIncentiveLevel(incentiveLevelCode)?.also {
```

Comment:
may revisit as if else might be clearer here

Reply:
yes id agree!